base/complex.jl

```diff
@@ -189,7 +189,11 @@ function show(io::IO, z::Complex)
     else
         print(io, compact ? "+" : " + ")
     end
-    show(io, i)
+    if isa(i,Signed) && !isa(i,BigInt) && i === typemin(typeof(i))
```

Comment:
Maybe move this condition up to the `i=-i`. One could imagine a checked integer type that throws an error on overflowing operations, so it would be nice not to actually run the `i=-i` in this case.

Reply:
That makes a lot of sense, I've updated the code accordingly.